File: tablet/Resources/menu/CatatoniaEpseMenu.js
// CatatoniaEpseMenu.js

/*
    Copyright (C) 2012-2015 Rudolf Cardinal (rudolf@pobox.com).
    Department of Psychiatry, University of Cambridge.
    Funded by the Wellcome Trust.

    This file is part of CamCOPS.

    Licensed under the Apache License, Version 2.0 (the "License");
    you may not use this file except in compliance with the License.
    You may obtain a copy of the License at

        http://www.apache.org/licenses/LICENSE-2.0

    Unless required by applicable law or agreed to in writing, software
    distributed under the License is distributed on an "AS IS" BASIS,
    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
    See the License for the specific language governing permissions and
    limitations under the License.
*/

/*jslint node: true, newcap: true */
"use strict";
/*global L */

module.exports = function CatatoniaEpseMenu() {

    var MenuWindow = require('menulib/MenuWindow'),
        UICONSTANTS = require('common/UICONSTANTS'),
        ALLTASKS = require('common/ALLTASKS'),
        data = [
            UICONSTANTS.CHANGE_PATIENT_MENU_LINE,
            ALLTASKS.TASKLIST.AIMS,
            // PERMISSION REFUSED: ALLTASKS.TASKLIST.BARS,
            // PERMISSION REFUSED: ALLTASKS.TASKLIST.BFCRS,
            {
                maintitle: L('t_catatoniaexam'),
                subtitle: L('s_catatoniaexam'),
                html: 'html/catatoniaexam.html'
            },
            // PERMISSION REFUSED: ALLTASKS.TASKLIST.CSI,
            // PERMISSION REFUSED: ALLTASKS.TASKLIST.GASS,
            // PERMISSION REFUSED: ALLTASKS.TASKLIST.LUNSERS,
            ALLTASKS.TASKLIST.MDS_UPDRS
            // PERMISSION REFUSED: ALLTASKS.TASKLIST.SAS,
        ],
        self = new MenuWindow({
            title: L('menutitle_catatonia_epse_b'),
            icon: UICONSTANTS.ICON_MENU_CATATONIA,
            data: data
        });

    return self;
};

File: tablet_qt/menu/catatoniaepsemenu.cpp
/*
    Copyright (C) 2012, University of Cambridge, Department of Psychiatry.
    Created by Rudolf Cardinal (rnc1001@cam.ac.uk).

    This file is part of CamCOPS.

    CamCOPS is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    CamCOPS is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with CamCOPS. If not, see <https://www.gnu.org/licenses/>.
*/

// #define INCLUDE_CATATONIA_EXAMINATION

#include "catatoniaepsemenu.h"

#include "common/uiconst.h"
#include "lib/uifunc.h"
#include "menulib/menuitem.h"

#ifdef INCLUDE_CATATONIA_EXAMINATION
    #include "common/urlconst.h"
#endif

#include "tasks/aims.h"
#include "tasks/mdsupdrs.h"

CatatoniaEpseMenu::CatatoniaEpseMenu(CamcopsApp& app) :
    MenuWindow(app, uifunc::iconFilename(uiconst::ICON_CATATONIA))
{
}

QString CatatoniaEpseMenu::title() const
{
    return tr("Catatonia and extrapyramidal side effects");
}

void CatatoniaEpseMenu::makeItems()
{
#ifdef INCLUDE_CATATONIA_EXAMINATION
    const QString examtitle = tr("Catatonia examination technique");
    const QString examsubtitle
        = tr("Standardized technique (for BFCRS, BFCSI)");
#endif
    m_items = {
        MAKE_CHANGE_PATIENT(m_app),
        MAKE_TASK_MENU_ITEM(Aims::AIMS_TABLENAME, m_app),
    // PERMISSION REFUSED: MAKE_TASK_MENU_ITEM("bars", m_app),
    // PERMISSION REFUSED: MAKE_TASK_MENU_ITEM("bfcrs", m_app),
#ifdef INCLUDE_CATATONIA_EXAMINATION
        MenuItem(
            examtitle,
            HtmlMenuItem(examtitle, urlconst::taskDocUrl("catatoniaexam")),
            "",
            examsubtitle
        ),
#endif
        // PERMISSION REFUSED: MAKE_TASK_MENU_ITEM("csi", m_app), // == bfcsi
        // PERMISSION REFUSED: MAKE_TASK_MENU_ITEM("gass", m_app),
        // PERMISSION REFUSED: MAKE_TASK_MENU_ITEM("lunsers", m_app),
        MAKE_TASK_MENU_ITEM(MdsUpdrs::MDS_UPDRS_TABLENAME, m_app),
        // PERMISSION REFUSED: MAKE_TASK_MENU_ITEM("sas", m_app),
    };
}
